### PR TITLE
Replace Mock.Of code with 10x faster equivalent

### PR DESCRIPTION
**Pull request by informatorius**
https://github.com/moq/moq4/issues/547

PR not finished yet ...

### CODE REVIEW

**Comment by stakx** [comment]
https://github.com/moq/moq4/blob/9c618125989d40f19bd8afa07ee21510b875e8d9/Source/Linq/Mock.cs#L53-L61

;-)

**Comment by informatorius** [comment]
Fixing ...
What about the benchmarks you did, should I include them somehow?
Or is this done with https://github.com/moq/moq4/issues/388

**Comment by stakx** [comment]
The time required to create a mock depends on the complexity of the type you want to mock. So including one single benchmark for one single mocked type wouldn't really give the full picture. All I wanted to see with that benchmark is an approximate speed difference (order of magnitude).

I'll review your PR a little later!

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ The format is loosely based on [Keep a Changelog](http://keepachangelog.com/en/1
 
 * Add `ISetupSequentialResult<TResult>.Returns` method overload that support delegate for deferred results (@snrnats, #594)
 
+#### Changed
+
+* Changed `Mock.Of<T>()` implementation to use much faster simple approach (@informatorius, #547)
```
Comment (stakx) [inline CHANGELOG.md]: In principle, this is a perfectly fine changelog entry. But may I humbly suggest that you shorten & rephrase it to be even more relevant from the user's perspective? For example:

> * Speed up `Mock.Of<T>()` by approx. one order of magnitude (...)

File: Source/Linq/Mock.cs [hunk 2]
```diff
@@ -57,7 +57,12 @@ public partial class Mock
 		/// <returns>The mocked object created.</returns>
 		public static T Of<T>() where T : class
 		{
-			return Mocks.CreateMockQuery<T>().First<T>();
+			//Disabled because of bad performance
+			//return Mocks.CreateMockQuery<T>().First<T>();
+			//and replaced by equivalent which is more than 10 times faster currently
```
Comment (stakx) [inline Source/Linq/Mock.cs]: Hmm, IMHO using the word "bad" isn't quite fair. I totally agree that the current implementation has *worse* performance than your new one; but does that make it *bad*?

I would've chosen a slightly different wording, for example:

```csharp
// This method was originally implemented as follows:
//
//     return Mocks.CreateMockQuery<T>().First<T>();
//
// which involved a lot of (unnecessary) `IQueryable` query provider overhead and
// lambda compilation. What it really boils down to is this (much faster) code:
...
```

File: CHANGELOG.md (hunk 1 above)
Comment (stakx) [inline CHANGELOG.md]: Please add one more blank line after this one.

Comment (stakx) [reply CHANGELOG.md]: Oh btw., please replace `#547` (issue) with `#598` (this PR).

**Comment by informatorius** [comment]
Done! Thanks for your perfect feedback.

**Comment by stakx** [comment]
Thank you for contributing this much faster reimplementation of `Mock.Of<T>()`! :+1:

**Comment by stakx** [comment]
@informatorius - I just pushed Moq 4.8.3 to NuGet, which includes your changes. Sorry it's taken so long.
